release_notes/ocp-4-0-release-notes.adoc

```diff
@@ -0,0 +1,715 @@
+of kernel tuning. The Node Tuning Operator provides a unified management
+interface to users of node-level sysctls and more flexibility to add custom
+tuning, which is currently a Technology Preview feature, specified by user
+needs. The Operator manages the containerized tuned daemon for OpenShift
+Container Platform as a Kubernetes DaemonSet. It ensures the custom tuning
+specification is passed to all containerized tuned daemons running in the
+cluster in the format that the daemons understand. The daemons run on all nodes
+in the cluster, one per node.
+
+[[ocp-4-0-metrics-and-logging]]
+=== Metrics and logging
+
+[[ocp-4-0-metering]]
+=== Metering
+
+[[ocp-4-0-metering-and-chargeback]]
+==== Metering and chargeback (Technology Preview)
+
+Metering and chargeback is a Technology Preview feature in {product-title} 4.0.
+
+The Operator Metering portion of the Operator framework gives administrators
+insight into usage for specific Operators. Usage metrics are unique to each
+Operator; metering watches custom resources and collects data, and performs
+calculations and reports on that data back to the administrator.
+
+[[ocp-4-0-cluster-monitoring]]
+==== Cluster monitoring
+
+{product-title} 4.0 provides a single, unified, and native experience for
+cluster administrators to manage alerts and start troubleshooting. It also
+provides a streamlined experience to set up app monitoring with Prometheus for
+tenants through the Operator Hub.
+
+[[ocp-4-0-autoscake-pods-horizontally-based-on-prometheus-metrics]]
+==== Autoscale pods horizontally based on any metrics in Prometheus (Technology Preview)
+
+This feature, currently in Technology Preview, enables you to configure
+horizontal pod autoscaling (HPA) based on any metric provided by Prometheus.
+Autoscale based on any cluster-level metrics {product-title} provides out of the
+box and autoscale based on any app metrics.
+
+Limitations:
+
+The adapter only connects to a single Prometheus instance (or a set of
+load-balanced replicas, using Kubernetes services). Manually deploying adapter
+and configuring it to use Prometheus (that includes Prometheus instance shipped
+in OpenShift) For using cluster-metrics, a cluster administrator needs to
+whitelist metrics that can be used to configure HPA.
+
+[[ocp-4-0-developer-experience]]
+=== Developer experience
+
+[[ocp-4-0-code-ready-containers]]
+==== Code ready containers
+
+A local desktop instance of {product-title} 4.0 replaces the functions of `oc
+cluster` commands, Minishift, and CDK. {product-title} 4.0 focuses on ease of
+access and native experience, with a native installer on macOS and Microsoft
+Windows, native hypervisor support, and tray icon integration.
+
+[[ocp-4-0-registry]]
+=== Registry
+
+`Red Hat Quay` container and application registry proves secure storage,
+distribution and deployment of containers in Red Hat {product-title} 4.0, and
+replaces `Cockpit` as the registry component.
```

Comment:
this is a little confusing as it sort of sounds like quay is used as the internal registry for the openshift 4.0 cluster (which it is not).  please work w/ PM (dirk herrmann)  to clarify this..i'm not sure what the intention was.

Reply:
@bparees Thank you. Will do.

Reply:
No idea where this sentence is coming from. But as Ben said it's wrong.

Reply:
Thanks! It was incorrectly sourced from the last roadmap presentation. This is now removed.